.github/workflows/rust.yml

```diff
@@ -84,4 +84,19 @@ jobs:
           nix_path: nixpkgs=channel:nixos-unstable
       - name: Integration Tests with nix-shell
         run: nix-shell --command ./scripts/integrationtest.sh
-
+  nix-build_test:
+    name: Nix-Build Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          default: true
+      - uses: Swatinem/rust-cache@v1
+      - uses: actions/checkout@v2.4.0
+      - uses: cachix/install-nix-action@v15
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
```

Comment:
Can we test on stable? Would be nice to make sure it actually works for users not using unstable.